### PR TITLE
Set warfile name to be artifactId

### PR DESCRIPTION
**Pull request by kandula-aswin**

### CODE REVIEW

**Review by jonathanaustin — COMMENTED** [review]
Not sure if this is the correct approach to rename the war to cater for a DevOps release pipeline.

I would think it is the responsibility of the release pipeline to handle setting the war context or "rename" the war. Renaming the war works for the current scenario of deploying to Tomcat, but what if we want to release it to an other container or have the need to use a completely different context? We would not want to rename the war in the build each time we need a new context.

I do feel it should be the responsibility of the release pipeline to handle the context a WAR file is deployed too. 

Also, even though we are not publishing this project to maven central, I am pretty sure the artifact would need to be the standard name with the version if we did publish, and as this is a template project, it could be used by projects that are published.

Let me know what you think.

**Comment by kandula-aswin** [comment]
Good point, I totally agree with you.

But my arguement is, it is not DevOps specific, but rather Tomcat specific. If you copy war file `"example-warfile.1.0.0.war"`. Then the context you visit will be `http://<host:port>/example-warfile.1.0.0` isn't it?

Looking at attachment_viewer, it defines something similar as per:
https://maven.apache.org/plugins-archives/maven-ear-plugin-2.8/examples/customizing-context-root.html

Another option is to define context for `"example-warfile.1.0.0.war"` in `<catalina_home>/conf/Catalina/localhost/ROOT.xml`
https://stackoverflow.com/questions/55448206/how-to-set-context-path-in-tomcat-8

**Comment by jonathanaustin** [comment]
Agree. It is tomcat specific behaviour. 

So as it is not a build artificat issue, probably best to leave it as the default name and handle the rename via the deploy mechanism or in the tomcat configuration.

**Comment by kandula-aswin** [comment]
* Giving `<warName>` does not impact publishing artifacts to maven repository
* war file name is defaults `${artifactId}`
* Default context name for applications deployed to web containers (Tomcat, JBoss) is war filename. So if `<warName>` is appropriately defined then it avoids container specific configuration

File: app-war/pom.xml [hunk 1]
```diff
@@ -30,6 +30,8 @@
 							<Ignore-Scanning-Archives>asm-all-repackaged-2.5.0.jar</Ignore-Scanning-Archives>
 						</manifestEntries>
 					</archive>
+					<!-- warName becomes the context of the application, it avoids container specific configuration -->
+					<warName>${artifactId}</warName>
```
Comment (jonathanaustin) [inline app-war/pom.xml]: What if we make this a property that defaults to the artifact+version and projects or (pipeline) can override it if they need to change the name.

File: pom.xml [hunk 2]
```diff
@@ -16,6 +16,8 @@
 		<failBuildOnCVSS>11</failBuildOnCVSS>
 		<!-- Update every 168 hours (7 days) -->
 		<cveValidForHours>168</cveValidForHours>
+		<app.warName>${project.build.finalName}</app.warName>
```
Comment (jonathanaustin) [inline pom.xml:20]: Might be good to include a brief comment here as well.

**Comment by kandula-aswin** [comment]
- Default build will ensure war name is per maven conventions
  - Example: `mvn package` the war name will be `${project.build.finalName}`
- War name can be changed by setting property `-Dapp.warName` and/or `-Dweb.warName`. The property can be set to any arbitrary value or to it's artificatId
  - arbitrary name: `mvn clean install -Dapp.warName=apple-war -Dweb.warName=orange-war`
  - artificatId name: `mvn clean install -Dapp.warName='${project.artifactId}' -Dweb.warName='${project.artifactId}'`

**Review by jonathanaustin — CHANGES_REQUESTED** [review]
Properties look good. Just made one request for a comment in the parent pom.

**Comment by jonathanaustin** [comment]
You should also add how to use the new properties in the README.

**Review by jonathanaustin — APPROVED** [review]
LGTM

**Comment by kandula-aswin** [comment]
Thank you Jonathan

**Pull request merged**